ibis/backends/pyspark/compiler.py

```diff
@@ -649,6 +649,27 @@ def compile_abs(t, expr, scope, timecontext, **kwargs):
     return F.abs(src_column)
 
 
+@compiles(ops.Clip)
+def compile_clip(t, expr, scope, timecontext, **kwargs):
+    op = expr.op()
+    col = t.translate(op.arg, scope, timecontext)
+    if op.upper is not None and op.lower is not None:
+        upper = t.translate(op.upper, scope, timecontext)
+        lower = t.translate(op.lower, scope, timecontext)
+        expr = F.when(col >= upper, F.lit(upper)).otherwise(
+            F.when(col <= lower, F.lit(lower)).otherwise(col)
```

Comment:
Happy with this implementation, but in case you haven't considered, you can implement an upper/lower bound in a (probably) simpler way using min and max. Like:
```python
def clip_lower(value, lower):
    return max(value, lower)
```
And equivalent for upper with min. I think your implementation here would be way simpler using this approach (and not sure if the translated expression could be faster too.

Reply:
@datapythonista Maybe I misunderstood your suggestion, but the object that we work with here are pyspark columns and sth like

```
max(spark_column, 0.0)
```

won't really work here.

Reply:
Or do you mean to implement like this?

```
def column_max(col1, col2):
    return when(col1 < col2, col2).otherwise(col1)
def column_min(col1, col2):
    return when(col1 < col2, col1).otherwise(col2)
def clip(col, lower, upper):
    return column_max(column_min(col, F.lit(upper)), F.lit(lower))
```

Reply:
Didn't use pyspark for a while and I don't remember, but isn't it a pyspark function to get the min or max between a column and a literal? Or if it's not, we should have it in Ibis, we could write this using them.

In any case, not important, if this is simpler, let's just go with this approach.

Reply:
I don't believe there is any native pyspark function that will directly get us this min/max comparison functionality. Probably easiest to leave it with the when/otherwise paradigm for now.